aws/logs_monitoring/lambda_function.py

```diff
@@ -852,6 +852,13 @@ def s3_handler(event, context, metadata):
             )
             yield structured_line
     else:
+        # CloudWatch Server Access Logs are double gzipped in some cases
```

Comment:
@sagar-khanna I'm sorry that we totally missed this PR somehow, and thanks for your contribution! Do you mind front-loading the logic within the `if` clause around line 844? I think it's clearer to handle decompression(s) all at once, unless I missed something. 

BTW, do you mind specify what kind of server access logs you are handling? I.e., "CloudWatch Server Access Logs" seems a little bit ambiguous to me.